src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -274,10 +284,40 @@ public ResponseEntity<UserDTO> getUser(@PathVariable String login) {
     @PreAuthorize("hasRole('ADMIN')")
     public ResponseEntity<Void> deleteUser(@PathVariable String login) {
         log.debug("REST request to delete User: {}", login);
+        if (isCurrentUser(login)) {
+            throw new BadRequestAlertException("You cannot delete yourself", "userManagement", "cannotDeleteYourself");
+        }
         userService.deleteUser(login);
         return ResponseEntity.ok().headers(HeaderUtil.createAlert(applicationName, "userManagement.deleted", login)).build();
     }
 
+    /**
+     * Delete users: deletes the provided users
+     *
+     * @param logins user logins to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("users")
+    @PreAuthorize("hasRole('ADMIN')")
+    public ResponseEntity<List<String>> deleteUsers(@RequestParam(name = "login") List<String> logins) {
+        log.debug("REST request to delete {} users", logins.size());
+        List<String> deletedUsers = new java.util.ArrayList<>();
+        for (String login : logins) {
+            try {
+                if (!isCurrentUser(login)) {
```

Comment:
This queries the database repeatedly for the same element due to `userRepository.getUser()`. I think removing the `currentUser.getLogin()` once before the loop from the list of `logins` should do the job as well. Then no more checks inside the loop are needed.

Reply:
Yes, that is absolutely correct - will change that accordingly